js-w2-hw.js

```diff
@@ -0,0 +1,129 @@
+// Flight booking fullname function
+
+function getFullname(firstName, lastName, useFormalName, gender) {
+  // Check if useFormalName is undefined
+  if (useFormalName === undefined) {
+    return "Error: Missing argument";
+  }
+
+  // Handle empty strings for firstName and lastName
+  if (!firstName && !lastName) {
```

Comment:
Interesting case checking for anonymous users. It's good that you made conditions to check for empty values, etc.